client/verta/verta/_internal_utils/_pip_requirements_utils.py

```diff
@@ -342,10 +345,12 @@ def inject_requirement(requirements, library, version):
                 if "==" in req:  # version pin: check version
                     their_ver = req.split('==')[-1]
                     if version != their_ver:  # versions conflict: raise exception
-                        raise ValueError(
-                            "Client is running with {} v{}, but the provided requirements specify v{};"
-                            " these must match".format(library, version, their_ver)
-                        )
+                        msg = ("Client is running with {} v{}, but the provided requirements specify v{};"
+                               " these should match").format(library, version, their_ver)
+                        if error_on_conflict:
+                            raise ValueError(msg)
+                        else:
+                            warnings.warn(msg, stacklevel=2)
```

Comment:
I didn't know about `stacklevel`! That's awesome. Other `warn()`s in the client could use that for sure 😅

Reply:
Yeah, I was irritated to see `warnings.warn(msg)` appended to the message and looked into it. It's slightly better this way. :)